### ISSUE TITLE
Convert directly from strings

### ISSUE THREAD

**Issue by jduranf** [ccoVeille/go-safecast#45]
To convert from any string to an int16, I have to do something like that:

```golang
func stringToInt16(numStr string) (int16, error) {
	numInt16, err := strconv.ParseInt(numStr, 10, 16)
	if err != nil {
		return 0, err
	}

	return int16(numInt16), nil
}
```

Falling in G115...

Why not add the option to convert directly from strings?

**Comment by ccoVeille** [comment]
Interesting idea, I also have to do such conversions.

I will take a look, but I'm unsure when.

I think the logic will require to check if there is a dot in it to see if it's a float, then try to convert to uint64 (or int64 if contains a dash), or convert to float64 if float. Then convert to the requested type

But yes, the logic seems simple ™️

**Comment by ccoVeille** [comment]
Here are some thoughts

I may support
- bool "true"/"false"
- octal 0644 and 0o644
- hexa 0xAA
- bite literal 0b111111
- negative value -1
- complex "2i"
- "snake thousands separator" 1_000
- float 1.23
- scientific notation 1e5 (check the syntax)
- [floating point literals](https://go.dev/ref/spec#Floating-point_literals) 

https://go.dev/ref/spec


<details>
<summary>
Some examples
</summary>
4_2
0600
0_600
0o600
0O600       // second character is capital letter 'O'
0xBadFace
0xBad_Face
0x_67_7a_2f_cc_40_c6
170141183460469231731687303715884105727
170_141183_460469_231731_687303_715884_105727

_42
0.
72.40
072.40       // == 72.40
2.71828
1.e+0
6.67428e-11
1E6
.25
.12345E+5
1_5.         // == 15.0
0.15e+0_2    // == 15.0

0x1p-2       // == 0.25
0x2.p10      // == 2048.0
0x1.Fp+0     // == 1.9375
0X.8p-0      // == 0.5
0X_1FFFP-16  // == 0.1249847412109375
</details>


This list is more about adding these to the units tests

**Comment by ccoVeille** [comment]
Available in [v1.4.0](https://github.com/ccoVeille/go-safecast/releases/tag/v1.4.0)
